### PR TITLE
Video events

### LINKED ISSUE #606: Pointer is behind video
First of all, pdfpc is amazing! I just discovered that it can show videos on my slides. However, I noticed a small issue with the pointer and video display (Surprisingly, the drawing tool does not exhibit any similar issues).

**Expected Behaviour**

Pointer is showing on top of a video 

**Actual Behaviour**

Pointer is hidden behind video 

**Steps to reproduce**

1. Download and extract https://pdfpc.github.io/demo/pdfpc-video-example.zip
2. Run pdfpc video-example.pdf
3. Navigate to a slide with a video
4. Activate "Laser" Pointer and move it over the video

pdfpc version: v4.5
Used Distribution (GTK version, vala version): GTK 3.24.30-2, Vala 0.52.5-1 (Arch Linux on kernel 5.13.12)
Display Server (X/Wayland): X
HiDPI Setting (yes/no): no

### PR DESCRIPTION
**Pull request by fnevgeny**
* Fix #606;
* Implement #348 (not 100%, which would be a lot of work, just for the current slide);
* Use different cursors when the pointer is above the video frame - play/stop/drag.

* Q1: The cursors (and the respective actions) are enabled in all modes except the drawing (pen/eraser) one. The idea is that e.g., while in the "laser pointer" mode, the video can still be controlled without switching the mode. On the other hand, the cursors may be annoying. Maybe restrict the video controls only to the "normal" mode?
* Q2: Maybe the colors in the play/stop cursors should be swapped (black outline, light fill)?

### CODE REVIEW

File: src/classes/action/link_action.vala [hunk 1]
```diff
@@ -63,10 +63,17 @@ namespace pdfpc {
             return new_obj as ActionMapping;
         }
 
+        protected override void on_mouse_enter(Gtk.Widget widget, Gdk.EventMotion event) {
```
Comment (AndreasBilke) [inline src/classes/action/link_action.vala:66]: Why do you switch visibility level for an overridden method?

File: src/classes/action/movie.vala [hunk 2]
```diff
@@ -42,6 +42,40 @@ namespace pdfpc {
         public Window.Fullscreen window { get; set; }
     }
 
+    protected struct PlaybackOptions {
+        /**
+         * A flag to indicate whether the movie should start automatically.
+         */
+        bool autostart;
```
Comment (AndreasBilke) [inline src/classes/action/movie.vala:49]: Is this flag in preparation for the (non working) autoplay feature (as mentioned within the man page)

**Comment by AndreasBilke** [comment]
Q1: I would disable video controls for all modes except normal mode. In my opinion it is counter intuitive to have modes where it works and some where it doesn't and I think this can lead to unwilling movie control clicking.

Q2: They should look like other icons, but in general I don't care

File: src/classes/action/link_action.vala (hunk 1 above)
Comment (fnevgeny) [reply src/classes/action/link_action.vala:66]: Well, I see no reason for them to be callable directly. Is it a problem?

File: src/classes/action/movie.vala (hunk 2 above)
Comment (fnevgeny) [reply src/classes/action/movie.vala:49]: The same structure is also used for the "href" style movie links, where autoplay does work (perhaps the main reason we still support it...)

**Comment by fnevgeny** [comment]
> Q1: I would disable video controls for all modes except normal mode. In my opinion it is counter intuitive to have modes where it works and some where it doesn't and I think this can lead to unwilling movie control clicking.

OK. Then probably it should be implemented at the more general level in `View.Behaviour.PdfLink` - the same logic should apply to clickable links as well.

File: src/classes/action/link_action.vala (hunk 1 above)
Comment (AndreasBilke) [reply src/classes/action/link_action.vala:66]: No, but in `ActionMapping` the virtual function is `public`.

Comment (fnevgeny) [reply src/classes/action/link_action.vala:66]: It's called from `View.Behaviour.PdfLink`, so it must be `public`. In general, there is quite some mess with `ActionMapping` returning copies of the class...

**Review by AndreasBilke — APPROVED** [review]

**Pull request merged**